core/src/main/java/org/elasticsearch/index/query/HasChildQueryParser.java

```diff
@@ -33,7 +33,12 @@
  */
 public class HasChildQueryParser implements QueryParser<HasChildQueryBuilder> {
 
-    private static final ParseField QUERY_FIELD = new ParseField("query", "filter");
+    public static final ParseField QUERY_FIELD = new ParseField("query", "filter");
+    public static final ParseField TYPE_FIELD = new ParseField("child_type", "type");
```

Comment:
Interesting, the docs for HasParentQuery have "parent_type" as the documented name. Maybe another thing to sort out for more consistency.